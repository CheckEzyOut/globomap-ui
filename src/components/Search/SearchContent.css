/*
Copyright 2019 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.search-content,
.search-content-base,
.search-content-results,
.search-content-results th,
.search-content-results td,
.search-content-results tbody {
  box-sizing: border-box;
}

.search-content {
  position: fixed;
  left: 256px;
  bottom: 0;
  right: 0;
  padding: 20px 7% 0;
  background-color: #fff;
  overflow-y: auto;
  top: 0;

  transform: translateY(250px);
  z-index: 7;
  transition: all 400ms cubic-bezier(0.215, 0.61, 0.355, 1);
}

.search-content.active {
  transform: translateY(0);
  z-index: 9;
}

.search-content-btn-close {
  border: 0;
  background: none;
  position: absolute;
  top: 20px;
  right: 20px;
  font-size: 1.6em;
  cursor: pointer;
  color: #ccc;
  outline: none;
}

.search-content-btn-close:hover {
  opacity: 0.75;
}

.search-content-results {
  position: relative;
  height: 100%;
  border-spacing: 0;
}

.search-content-results thead,
.search-content-results tfoot,
.search-content-results tbody tr {
  display: table;
  width: 100%;
  table-layout: fixed;
}

.search-content-results thead {
  border-bottom: 4px solid rgba(0, 0, 0, 0.05);
}

.search-content-results thead th {
  border-bottom: 1px solid #333;
  height: 50px;
}

.search-content-results tfoot {
  background-color: white;
}

.search-content-results tfoot th {
  height: 60px;
}

.search-content-results tbody {
  position: absolute;
  left: 0;
  right: 0;
  top: 50px;
  bottom: 60px;
  padding: 10px 0;
  overflow-y: auto;
  overflow-x: hidden;
}

.search-content-results .row-tools-head,
.search-content-results .row-tools-body {
  width: 120px;
}

.search-content-results .type-head,
.search-content-results .type-body {
  width: 25%;
}

.search-content-results tr td {
  position: relative;
  cursor: pointer;
  font-size: 0.95em;
  color: #666;
}

.search-content-results tr:hover td {
  color: #2eb78e;
}

.search-content-results tr:hover button {
  color: #2eb78e;
  opacity: 0.5;
}

.search-content-results th,
.search-content-results td {
  text-align: left;
  color: #222;
  font-size: 1em;
  border-bottom: 1px solid #f2f2f2;
}

.search-content-results th {
  font-weight: bold;
  font-size: 1.2em;
}

.search-content-results tr:hover .row-tools {
  display: flex;
}

.search-content-results .row-tools button {
  padding: 8px 8px;
  font-size: 1.0em;
  background: none;
  margin-left: 0 10px;
  border: none;
  opacity: 0.05;
  width: 48px;
}

.search-content-results .row-tools button:hover {
  opacity: 1;
}

.search-content-base {
  position: absolute;
  bottom: 0;
  display: flex;
  align-items: center;
  height: 60px;
  border-top: 1px solid rgba(0, 0, 0, 0.3)
}
