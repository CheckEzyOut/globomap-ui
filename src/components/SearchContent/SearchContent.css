/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.search-content,
.search-content-base,
.search-content-results {
  box-sizing: border-box;
}

.search-content {
  position: absolute;
  top: 85px;
  left: 0;
  bottom: 0;
  width: 70%;
  /*border-radius: 0 0 4px 4px;*/
  overflow: hidden;
  /*box-shadow: 0 1px 2px rgba(0,0,0,0.2);*/
}

.search-content.with-info {

}

.search-content-results {
  width: 100%;
  border-spacing: 0;
}

.search-content-results thead,
.search-content-results tbody tr {
  display: table;
  width: 101%;
  table-layout: fixed;
}

.search-content-results tbody {
  position: absolute;
  left: 0;
  right: 0;
  top: 31px;
  bottom: 32px;
  overflow-y: auto;
  overflow-x: hidden;
}

.search-content-results th:first-child,
.search-content-results td:first-child {
  width: 10%;
}

.search-content-results th:nth-child(2),
.search-content-results td:nth-child(2) {
  width: 20%;
}

.search-content-results tr td {
  cursor: pointer;
}

.search-content-results tr:nth-child(odd) {
  /*background-color: #efefef;*/
}

.search-content-results tr.current td,
.search-content-results tr.current:hover td {
  background-color: #3872d2;
  color: white;
}

.search-content-results tr:hover td {
  background-color: rgba(56, 114, 210, 0.1);
}

.search-content-results th {
  background-color: #f8f8f8;
}

.search-content-results th,
.search-content-results td {
  text-align: left;
  padding: 8px 14px;
  color: #454545;
  font-size: 13px;
}

.search-content .btn-open-node {
  font-size: 16px;
}

.search-content-base {
  display: flex;
  position: absolute;
  padding: 4px;
  left: 0;
  bottom: 0;
  right: 0;
  height: 32px;
  background-color: #e2e2e2;
}
