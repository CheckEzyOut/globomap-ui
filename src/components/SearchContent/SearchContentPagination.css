/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.search-content-pagination {
  box-sizing: border-box;
}

.search-content-pagination {
  display: flex;
}

.search-content-pagination .page-number {
  text-align: center;
  width: 60px;
  border-radius: 0;
  border-size: 1px 0 1px 0;
  box-shadow: none;
}

.search-content-pagination .btn-previous {
  border-right: 0;
  border-radius: 4px 0 0 4px;
}

.search-content-pagination .btn-next {
  border-left: 0;
  border-radius: 0 4px 4px 0;
}

.search-content-pagination .pagination-info {
  display: flex;
  align-items: center;
  font-size: 12px;
  color: white;
  margin: 0 15px;
}
