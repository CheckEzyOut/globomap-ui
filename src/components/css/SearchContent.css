.search-content {
  box-sizing: border-box;
}

.search-content {
  position: absolute;
  top: 110px;
  width: 100%;
  left: 0;
}

.search-content table {
  width: 100%;
  border-spacing: 0;
}

.search-content table tr td {
  cursor: pointer;
}

.search-content table tr:nth-child(odd) {
  background-color: #efefef;
}

.search-content table tr.current td,
.search-content table tr.current:hover td {
  background-color: #9daca9;
  color: white;
}

.search-content table tr:hover td {
  background-color: #dbeae7;
}

.search-content table th {
  background-color: #d9dddd;
}

.search-content table th,
.search-content table td {
  text-align: left;
  padding: 6px 10px;
  color: #454545;
  font-size: 14px;
}

.search-content .btn-open-node {
  font-size: 16px;
}
