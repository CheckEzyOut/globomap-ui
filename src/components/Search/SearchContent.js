/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  clearNodes,
  setCurrentNode,
  clearCurrentNode
} from '../../redux/modules/nodes';
import { addStageNode } from '../../redux/modules/stage';
import { setTab } from '../../redux/modules/tabs';
import SearchContentPagination from './SearchContentPagination';
import { Properties } from '../';
import './SearchContent.css';

class SearchContent extends Component {

  constructor(props) {
    super(props);
    this.onNodeSelect = this.onNodeSelect.bind(this);
  }

  onNodeSelect(event, node) {
    event.stopPropagation();
    if (this.props.currentNode !== null &&
        this.props.currentNode._id === node._id) {
      return this.props.clearCurrentNode();
    }
    return this.props.setCurrentNode(node);
  }

  onOpenMap(event, node) {
    event.stopPropagation();
    this.props.addStageNode(node, null, true);
    // change to map screen
    this.props.setTab('map');
  }

  onOpenProperties(event, node) {
    event.stopPropagation();
  }

  onCloseSearchContent(event) {
    event.stopPropagation();
    this.props.clearNodes();
    this.props.clearCurrentNode();
  }

  render() {
    let index = this.props.perPage * (this.props.currentPage - 1);
    index = (index === 0 ? 1 : index + 1);

    let allNodes = [];
    if (this.props.nodeList !== undefined) {
      allNodes = this.props.nodeList.map((node, i) => {
        let current = (this.props.currentNode &&
                       node._id === this.props.currentNode._id);
        return (
          <React.Fragment key={node._id}>
            <tr className={current ? 'current' : ''}
                onClick={e => this.onNodeSelect(e, node)}>
              <td>{i + index}</td>
              <td>{this.props.namedCollections[node.type].alias}</td>
              <td>{node.name}</td>
            </tr>
            <tr className={`node-options${current ? ' open' : ''}`}>
              <td colSpan="3">
                <button onClick={e => this.onOpenProperties(e, node)}>
                  <i className="icon fa fa-info" /> Properties
                </button>
                <button onClick={e => this.onOpenMap(e, node)}>
                  <i className="icon fa fa-sitemap" /> View Map
                </button>
                {/* <div className="node-properties">
                  <Properties item={node} />
                </div> */}
              </td>
            </tr>
          </React.Fragment>
        );
      });
    }

    return (
      <div className={`search-content${allNodes.length > 0 ? ' active' : ''}`}>
        <button className="search-content-btn-close"
                onClick={e => this.onCloseSearchContent(e)}>
          <i className="fa fa-times"></i>
        </button>
        {allNodes.length > 0 &&
          <table className="search-content-results">
            <thead>
              <tr>
                <th>#</th>
                <th>Type</th>
                <th>Name</th>
              </tr>
            </thead>
            <tbody>{allNodes}</tbody>
            <tfoot className="search-content-base">
              <tr>
                <th><SearchContentPagination /></th>
              </tr>
            </tfoot>
          </table>}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    nodeList: state.nodes.nodeList,
    currentNode: state.nodes.currentNode,
    perPage: state.nodes.perPage,
    currentPage: state.nodes.currentPage,
    namedCollections: state.app.namedCollections
  };
}

export default connect(
  mapStateToProps,
  { addStageNode, setCurrentNode, clearNodes,
    clearCurrentNode, setTab }
)(SearchContent);
