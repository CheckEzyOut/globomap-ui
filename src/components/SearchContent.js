/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import SearchContentPagination from './SearchContentPagination';
import './css/SearchContent.css';

class SearchContent extends Component {

  constructor(props) {
    super(props);
    this.onNodeSelect = this.onNodeSelect.bind(this);
  }

  render() {
    let index = 1;
    if (this.pagination) {
      index = this.pagination.state.total * (this.pagination.state.pageNumber - 1);
      index = (index === 0 ? 1 : index + 1);
    }

    if (!this.props.nodes) {
      return null;
    }

    let allNodes = this.props.nodes.map((node, i) => {
      let current = (this.props.currentNode &&
                     node._id === this.props.currentNode._id);

      return <tr key={node._id} className={current ? 'current' : ''}
                 onClick={(e) => this.onNodeSelect(e, node)}>
              <td>{i + index}</td>
              <td>{node.type}</td>
              <td>{node.name}</td>
             </tr>;
    });

    return (
      <div className={'search-content' + (this.props.currentNode ? ' with-info' : '')}>
        <table className="search-content-results">
          <thead>
            <tr>
              <th>#</th>
              <th>Type</th>
              <th>Name</th>
            </tr>
          </thead>
          <tbody>{allNodes.length > 0 && allNodes}</tbody>
        </table>
        <div className="search-content-base">
          <SearchContentPagination
            ref={(pagination) => {this.pagination = pagination}}
            nodes={this.props.nodes}
            findNodes={this.props.findNodes}
            enabledCollections={this.props.enabledCollections}
            header={this.props.header} />
        </div>
      </div>
    );
  }

  onNodeSelect(event, node) {
    event.stopPropagation();
    this.props.addNodeToStage(node, false, true);
  }

}

export default SearchContent;
