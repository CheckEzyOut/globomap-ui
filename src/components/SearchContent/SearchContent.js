/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { traversalSearch } from '../../redux/modules/nodes';
import { addStageNode } from '../../redux/modules/stage';
import { Loading } from '../';
import SearchContentPagination from './SearchContentPagination';
import './SearchContent.css';

class SearchContent extends Component {

  constructor(props) {
    super(props);
    this.onNodeSelect = this.onNodeSelect.bind(this);
  }

  onNodeSelect(event, node) {
    event.stopPropagation();
    this.props.addStageNode(node, null, true);
    if (this.props.currentNode === null) {
      this.props.traversalSearch({ node });
    }
  }

  render() {
    let index = this.props.perPage * (this.props.currentPage - 1);
    index = (index === 0 ? 1 : index + 1);

    let allNodes = [];
    if (this.props.nodeList !== undefined) {
      allNodes = this.props.nodeList.map((node, i) => {
        let current = (this.props.currentNode &&
                       node._id === this.props.currentNode._id);
        return (
          <tr key={node._id} className={current ? 'current' : ''}
              onClick={(e) => this.onNodeSelect(e, node)}>
            <td>{i + index}</td>
            <td>{this.props.namedCollections[node.type].alias}</td>
            <td>{node.name}</td>
          </tr>
        );
      });
    }

    return (
      <div className="search-content">
        {allNodes.length > 0 && allNodes &&
          <table className="search-content-results">
            <thead>
              <tr>
                <th>#</th>
                <th>Type</th>
                <th>Name</th>
              </tr>
            </thead>
            <tbody>{allNodes}</tbody>
          </table>}
        <div className="search-content-base">
          <SearchContentPagination />
        </div>
        <Loading isLoading={this.props.findLoading} iconSize="big" />
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    nodeList: state.nodes.nodeList,
    currentNode: state.nodes.currentNode,
    perPage: state.nodes.perPage,
    findLoading: state.nodes.findLoading,
    currentPage: state.nodes.currentPage,
    namedCollections: state.app.namedCollections
  };
}

export default connect(
  mapStateToProps,
  { traversalSearch, addStageNode }
)(SearchContent);
