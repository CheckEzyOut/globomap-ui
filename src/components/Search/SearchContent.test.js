import React from 'react';
import { shallow } from 'enzyme';
import { SearchContent } from './SearchContent';

describe("SearchContent", function() {

  it('should render SearchContent component', () => {
    const component = shallow(<SearchContent />);
    expect(component).toMatchSnapshot();
  });

});
