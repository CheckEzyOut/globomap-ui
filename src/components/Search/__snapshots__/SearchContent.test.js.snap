// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContent should render SearchContent component 1`] = `
<div
  className="search-content"
>
  <button
    className="search-content-btn-close"
    onClick={[Function]}
  >
    <i
      className="fa fa-times"
    />
  </button>
</div>
`;
